livestream-compose-sample/src/main/java/io/getstream/livestream/compose/ui/LiveStreamHomeActivity.kt

```diff
@@ -0,0 +1,120 @@
+import androidx.compose.ui.unit.dp
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.api.models.QuerySort
+import io.getstream.chat.android.client.models.Filters
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.viewmodel.channel.ChannelListViewModel
+import io.getstream.chat.android.compose.viewmodel.channel.ChannelViewModelFactory
+import io.getstream.chat.android.offline.ChatDomain
+import io.getstream.livestream.compose.R
+import io.getstream.livestream.compose.darkColorPalette
+import io.getstream.livestream.compose.lightColorPalette
+
+class LiveStreamHomeActivity : ComponentActivity() {
+    private val factory by lazy {
+        ChannelViewModelFactory(
+            ChatClient.instance(), ChatDomain.instance(),
+            QuerySort.desc("last_updated"),
+            Filters.and(
+                Filters.eq("type", "livestream"),
+                Filters.`in`("members", listOf(ChatClient.instance().getCurrentUser()?.id ?: ""))
+            )
+        )
+    }
+
+    private val listViewModel: ChannelListViewModel by viewModels { factory }
+
+    @ExperimentalMaterialApi
+    @ExperimentalFoundationApi
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            var expanded by remember { mutableStateOf(false) }
+            var isDarkMode by remember { mutableStateOf(false) }
+            var isGrid by remember { mutableStateOf(true) }
+
+            ChatTheme(colors = if (isDarkMode) darkColorPalette() else lightColorPalette()) {
+                CustomHeader(
+                    title = resources.getString(R.string.app_name),
+                    actions = {
+                        //Adds a icon button for switching theme
+                        IconButton(
+                            onClick = {
+                                expanded = !expanded
+                            }) {
+                            Image(
+                                painter = painterResource(id = R.drawable.ic_theme_switch),
+                                contentDescription = stringResource(id = R.string.accessibilitySwitchTheme),
+                                modifier = Modifier
+                                    .width(24.dp)
+                                    .height(24.dp)
+                            )
+                        }
+                        //Theme chooser drop down selector
+                        DropdownMenu(
+                            expanded = expanded,
+                            onDismissRequest = { expanded = false }
+                        ) {
+
+                            DropdownMenuItem(onClick = {
+                                expanded = !expanded
+                                isDarkMode = false
+                            }) {
+                                Text(text = "LiveStream light theme")
```

Comment:
pull out to resources, or maybe have a list of values or something

Reply:
Updated